client/src/nv_ingest_client/client/interface.py

```diff
@@ -66,6 +72,74 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+class LazyLoadedList:
+    def __init__(self, filepath: str, expected_len: Optional[int] = None):
+        self.filepath = filepath
+        self._len = expected_len  # Store pre-calculated length
+
+    def __iter__(self) -> Iterator[Any]:
+        try:
+            with open(self.filepath, "r", encoding="utf-8") as f:
+                for line in f:
+                    yield json.loads(line)
+        except FileNotFoundError:
+            logger.error(f"LazyLoadedList: File not found {self.filepath}")
+            return iter([])
+        except json.JSONDecodeError as e:
+            logger.error(f"LazyLoadedList: JSON decode error in {self.filepath} during iteration: {e}")
+            raise
+
+    def __len__(self) -> int:
+        if self._len is not None:
+            return self._len
+        try:
+            with open(self.filepath, "r", encoding="utf-8") as f:
+                self._len = sum(1 for _ in f)
+            return self._len
+        except FileNotFoundError:
+            self._len = 0
+            return 0
+        except Exception as e:
+            logger.error(f"LazyLoadedList: Error calculating len for {self.filepath}: {e}")
+            self._len = 0
+            return 0
+
+    def __getitem__(self, idx: int) -> Any:
```

Comment:
This is going to be pretty inefficient for multiple indexing calls with large files. Something that would make this more performant would be either pre-computed or lazy offset cache.

Pre-compute:
```python
def _build_index(self):
    self._offsets = []
    with open(self.filepath, "rb") as f:
        pos = f.tell()
        for line in f:
            self._offsets.append(pos)
            pos += len(line)

 def __getitem__(self, idx):
    if not self._offsets:
        self._build_index()

    with open(self.filepath, "rb") as f:
        f.seek(self._offsets[idx])
        return json.loads(f.readline().decode("utf-8"))
        
      ... snip ...
```

Reply:
Also might make sense to just build the index at init time.